pkg/jsonSchemaValidator/validator.go

```diff
@@ -12,18 +19,204 @@ func New() *JSONSchemaValidator {
 	return &JSONSchemaValidator{}
 }
 
-type Result = gojsonschema.Result
+type resourceMinimumCompiler struct{}
+type resourceMaximumCompiler struct{}
+
+type resourceMinimumSchema string
+type resourceMaximumSchema string
 
-func (jsv *JSONSchemaValidator) ValidateYamlSchema(schemaContent string, yamlContent string) (*Result, error) {
+func (jsv *JSONSchemaValidator) ValidateYamlSchema(schemaContent string, yamlContent string) ([]jsonschema.Detailed, error) {
 	jsonSchema, _ := yaml.YAMLToJSON([]byte(schemaContent))
 	return jsv.Validate(string(jsonSchema), yamlContent)
 }
 
-func (jsv *JSONSchemaValidator) Validate(schemaContent string, yamlContent string) (*Result, error) {
-	jsonContent, _ := yaml.YAMLToJSON([]byte(yamlContent))
+func (jsv *JSONSchemaValidator) Validate(schemaContent string, yamlContent string) ([]jsonschema.Detailed, error) {
+	var m interface{}
+	err := yaml.Unmarshal([]byte(yamlContent), &m)
+
+	if err != nil {
+		panic(err)
+	}
+
+	m, err = toStringKeys(m)
+	if err != nil {
+		panic(err)
+	}
+
+	compiler := jsonschema.NewCompiler()
+	compiler.AssertFormat = true
+
+	if err := compiler.AddResource("schema.json", strings.NewReader(schemaContent)); err != nil {
+		panic(err)
```

Comment:
I don't think we want to panic here, this will alert pagerduty in case the user passed in an invalid yaml file!

Reply:
perhaps return an error instead and handle it in the parent functions?